### ISSUE TITLE
Expand `non_topologically_sorted_functions` lint warnings

### ISSUE THREAD

**Issue by smoelius** [trailofbits/dylint#1886]
https://github.com/trailofbits/dylint/pull/1799 added a most awesome lint to flag modules with function definitions that are not topologically sorted.

However, for cases like this, its [warning](https://github.com/trailofbits/dylint/blob/bcd968f8c418d1d18b1ac103e3a6cc823c5d0bc6/examples/restriction/non_topologically_sorted_functions/ui/multiple_incorrect.stderr) could be more informative: https://github.com/trailofbits/dylint/blob/bcd968f8c418d1d18b1ac103e3a6cc823c5d0bc6/examples/restriction/non_topologically_sorted_functions/ui/multiple_incorrect.rs#L3-L15
```
warning: 
  --> $DIR/multiple_incorrect.rs:15:1
   |
LL | fn bar() {}
   | ^^^^^^^^^^^ function `bar` should be defined before `baz`
   |
   = help: move bar's definition to earlier in the module
   = note: `#[warn(non_topologically_sorted_functions)]` on by default

warning: 1 warning emitted
```
The lint is flagging `bar` because `bar` is called before `baz` in `foo`.

The warning could say this (or something similar).

**Comment by smoelius** [comment]
Fixed by #1950
